src/components/NavMenu.tsx

```diff
@@ -0,0 +1,312 @@
+import React, { useRef, useState, useEffect, useCallback } from 'react';
+import {
+	SafeAreaView,
+	View,
+	FlatList,
+	Text,
+	Image,
+	TouchableOpacity,
+	Animated,
+	PanResponder,
+	Easing,
+	StyleSheet,
+	Dimensions,
+	ImageSourcePropType,
+} from 'react-native';
+import { Icon } from 'react-native-elements';
+
+import Toggle from './Toggle';
+
+interface HeaderProps {
+	imgSrc: ImageSourcePropType;
+	title: string;
+	subtitles: string[];
+}
+
+const MenuHeader = ({ imgSrc, title, subtitles }: HeaderProps) => {
+	return (
+		<View style={styles.menuHeader}>
+			<Image source={imgSrc} style={imageStyles.headerImage} />
+			<View>
+				<Text style={styles.text}>{title}</Text>
+				{subtitles.map((sub) => (
+					<Text style={{ fontSize: 12 }} key={sub}>
+						{sub}
+					</Text>
+				))}
+			</View>
+		</View>
+	);
+};
+
+interface Link {
+	text: string;
+	route: string;
+	iconName: string;
+}
+
+interface MenuProps {
+	header: HeaderProps;
+	links: Link[];
+	widthPercent: number;
+	toggleState: boolean;
```

Comment:
Instead of abstracting the toggle controls out, we can just define its function directly in this component since I don't think we're going to be using the toggle for anything else.